addons/voice/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleCloudAPI.java

```diff
@@ -261,41 +318,27 @@ private void saveAudioAndTextToFile(String text, File cacheFile, byte[] audio, S
      *
      * @param text Text to synthesise
      * @param voice Voice parameter
+     * @param audioFormat Audio encoding format
      * @return Audio input stream
      */
-    private byte[] synthesizeSpeechByGoogle(String text, GoogleTTSVoice voice, String audioFormat) {
-        // Set the text input to be synthesized
-        SynthesisInput.Builder builder = SynthesisInput.newBuilder();
-        if (text.startsWith("<speak>")) {
-            builder.setSsml(text);
-        } else {
-            builder.setText(text);
-        }
-        SynthesisInput input = builder.build();
-
-        // Build the voice request
-        VoiceSelectionParams voiceParam = VoiceSelectionParams.newBuilder()
-                .setLanguageCode(voice.getLocale().getLanguage())
-                .setName(voice.getLabel())
-                .setSsmlGender(SsmlVoiceGender.forNumber(voice.getSsmlGender()))
-                .build();
-
-        // Select the type of audio file you want returned
-        AudioConfig audioConfig = AudioConfig.newBuilder()
-                .setAudioEncoding(AudioEncoding.valueOf(audioFormat)) // MP3 or LINEAR16 audio.
-                .setPitch(config.getPitch())
-                .setSpeakingRate(config.getSpeakingRate())
-                .setVolumeGainDb(config.getVolumeGainDb())
-                .build();
-
-        // Perform the text-to-speech request
-        SynthesizeSpeechResponse response = googleClient.synthesizeSpeech(input, voiceParam,
-                audioConfig);
-
-        // Get the audio contents from the response
-        ByteString audioContents = response.getAudioContent();
-
-        return audioContents.toByteArray();
+    private byte[] synthesizeSpeechByGoogle(String text, GoogleTTSVoice voice, String audioFormat) throws IOException {
+        AudioConfig audioConfig = new AudioConfig(AudioEncoding.valueOf(audioFormat), config.getPitch(),
+                config.getSpeakingRate(), config.getVolumeGainDb());
+        SynthesisInput synthesisInput = new SynthesisInput(text);
+        VoiceSelectionParams voiceSelectionParams = new VoiceSelectionParams(voice.getLocale().getLanguage(),
+                voice.getLabel(), SsmlVoiceGender.valueOf(voice.getSsmlGender()));
+
+        SynthesizeSpeechRequest request = new SynthesizeSpeechRequest(audioConfig, synthesisInput,
+                voiceSelectionParams);
+
+        HttpRequestBuilder builder = HttpRequestBuilder.postTo(SYTNHESIZE_SPEECH_URL)
+                .withHeader(AUTH_HEADER_NAME, getAuthorization()).withContent(gson.toJson(request), APPLICATION_JSON);
+
+        SynthesizeSpeechResponse synthesizeSpeechResponse = gson.fromJson(builder.getContentAsString(),
+                SynthesizeSpeechResponse.class);
+
+        byte[] encodedBytes = synthesizeSpeechResponse.getAudioContent().getBytes(StandardCharsets.UTF_8);
```

Comment:
Null check for `synthesizeSpeechResponse`?

Reply:
I've added some null checks which may cover some border cases when the response is null due to encoding errors. Because there are no null annotations on Gson the compiler starts showing warnings about unncessary null checks. I've suppressed these with the `@SuppressWarnings` annotations.